product_supplierinfo_purchase_contact/models/stock_rule.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2024 Tecnativa - Carolina Fernandez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, models
+
+
+class StockRule(models.Model):
+    _inherit = "stock.rule"
+
+    def _make_po_get_domain(self, company_id, values, partner):
+        if values.get("supplier") and values["supplier"].purchase_partner_id != partner:
+            partner = values["supplier"].purchase_partner_id
+        return super()._make_po_get_domain(company_id, values, partner)
+
+    def _prepare_purchase_order(self, company_id, origins, values):
+        res = super()._prepare_purchase_order(company_id, origins, values)
+        values = values[0]
+        if "supplier" in values and values["supplier"].purchase_partner_id:
+            res["partner_id"] = values["supplier"].purchase_partner_id.id
+        return res
+
+    @api.model
+    def _prepare_purchase_order_line(
+        self, product_id, product_qty, product_uom, company_id, supplier, po
+    ):
+        res = super()._prepare_purchase_order_line(
+            product_id, product_qty, product_uom, company_id, supplier, po
+        )
+        if supplier.purchase_partner_id:
+            partner = supplier.purchase_partner_id
+            uom_po_qty = product_uom._compute_quantity(
+                product_qty, product_id.uom_po_id
+            )
+            seller = product_id.with_company(company_id)._select_seller(
+                partner_id=partner,
+                quantity=uom_po_qty,
+                date=po.date_order and po.date_order.date(),
+                uom_id=product_id.uom_po_id,
+            )
+            res.update(self._prepare_purchase_order_line_from_seller(seller))
+        return res
```

Comment:
This method doesn't belong to this  model (is from purchase.order.line). Is it needed at all?

Reply:
Not necessary i just added to check if test failed, removed it